core/multigroup/multigroup_types.py

```diff
@@ -30,13 +30,15 @@ def create_multigroup(bm, faces, prop):
     """
 
     # Prevent error when there are no components
-    if len(str(prop.components)) == 0:
+    if len(prop.components) == 0:
         popup_message("No components are chosen", "No Components Error")
+        prop.components = "dw"
         return False
 
     # Prevent error when there are invalid chars
-    if not re.match("^[dw]*$", str(prop.components)):
+    if not re.match("^[dw]*$", prop.components):
         popup_message("Components cannot contain chars other than d and w", "Invalid Characters Error")
+        prop.components = re.sub("[^d|w|]", "", prop.components)
```

Comment:
If we have found invalid characters and removed them, it should be okay to continue with the procedure. No return necessary.
Also, since we are solving the problem, the popup message is not really relevant.